app-editors/vim/vim-8.2.0803.recipe

```diff
@@ -79,18 +74,29 @@ BUILD_PREREQUIRES="
 	cmd:gettext
 	cmd:grep
 	cmd:make
-#	cmd:perl
 	cmd:pkg_config$secondaryArchSuffix
 	cmd:python3
-	cmd:ruby
 	cmd:sed
 	"
+#	cmd:perl
+#	cmd:ruby
+
+
+PATCHES="
+	vim-8.2.0803.patchset
+	"
 
 BUILD()
 {
 
 # Global ----------------------------------------
 
+	# Override default libs for GCC2 (different C++ libraries).
+	if [ "$effectiveTargetArchitecture" == "x86_gcc2" ] ; then
+		export HAIKUGUI_LIBS1="-lbe -lroot -ltracker -ltranslation -lstdc++.r4"
```

Comment:
Messier to implement.  This way if you use the plain makefile, it will default to GCC4+ libraries.

Actually, I was wondering if I could check the compiler in the makefile, but that would take ages to figure out (the generic Haiku makefile has some code for getting the compiler version, but in the vim Makefile I don't know if it even knows what compiler it is using at that stage, and there is trouble running code in the definition section of the Makefile).  Hours of fiddling required.

Reply:
Maybe. I imained something like this: store the common libs in one variable, and question-mark the other with the differences, then add the to together and use for make.

Reply:
But i don't know why don't we just invert the original logic in the makefile. It checks for GCC version and if it is 8, then uses one lib, otherwise another. This needs adjustments time-to-time if the gcc package gets major-version bump (5->8, for example). Instead checking for the always changing version we should check the other way around, as GCC2 won't change anymore. So i would use the original implementation for this lib magic, but @bitgichi said it is not enough to get it built, if so, then it should be patched, not creating alternative logic layers.

Reply:
Previous if-logic in Makefile does not build with BSD make, therefore cannot stay upstream. If you know a way to make it work both with BSD and GNU make, feel free to send a patch upstream.

See: https://github.com/vim/vim/pull/5605#issuecomment-583884440

Reply:
After a couple of hours fiddling around with makefiles, I got one which tests the compiler version and sets the header appropriately.  Wonder if it works with BSD make, since it uses ifeq (a,b) and some $(shell ...) calls to get the compiler version.  At least my previous ?= patch was safe that way.  Though I see the autoconf stuff does get the compiler version too, so would the stdc libraries to link change fit better there somehow?

Reply:
@agmsmith, did the previous `?=` version build on gcc2 without HP?

IMHO, it's better to keep it platform/recipe-agnostic, sure, HP is a nice way to get software but not everyone should be required to use it, in order to install software.

Reply:
No, the ?= one would build only on later GCC.  Unless you set the environment variable to the magic value for GCC2.

Reply:
Can we use the `?=` version, and update [Vim build instructions](https://github.com/vim/vim/blob/master/runtime/doc/os_haiku.txt) for 32 bit Haiku to set the environment variable as needed? Do you think it is feasible?

Reply:
Yes, that could work as a stop-gap measure.  Shouldn't upset the vim people.

Would be better if someone learnt the configure system and added Haiku GCC version detection in there.  That would start with looking at src/configure.ac and https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Writing-configure_002eac.html  Then getting it to generate a variable containing the appropriate stdc++ library file names for the Makefile to use.  Hours of fiddling.

Reply:
Well, I don't have time to do the upstreaming (check BSD make compatibility, learn autoconf, talk to the vim people, etc), so the patch the Makefile method from my latest commit will work for now (and avoids upstreaming).  Time to get back to the Haiku package system first boot problem, and then get back to my less evil reputation based social media web site, then get back to updating the resume and getting a job!  Anyone else, feel free to advance this vim problem.